core/src/main/java/org/infinispan/configuration/cache/TracingConfigurationBuilder.java

```diff
@@ -73,6 +74,11 @@ public TracingConfigurationBuilder enableCategory(SpanCategory category) {
       return this;
    }
 
+   public void spanCategories(Stream<SpanCategory> spanCategories) {
```

Comment:
Instead of passing a `Stream` here, I think we should just pass `Set<SpanCategory>` as this is more consistent with other *Builder configuration classes. This means we need to add the `collect(Collectors.toSet())` to the CacheParser.